### PR TITLE
make bot robust for server/client dcs

### PR DESCRIPTION
**Pull request by leantOnSol**

### CODE REVIEW

File: main.ts [hunk 1]
```diff
@@ -82,7 +82,13 @@ async function runSetup() {
...
         const guild = await client.guilds.fetch(GUILD_ID);
         const channel = await guild.channels.fetch(CHANNEL_ID);
-
+        // On error, try to reconnect
+        socket.onerror = async (event) => {
+            console.log("WS connection error, sleeping, cleaning up and trying to reconnect again");
+            await sleep(1000);
+            cleanUp();
+            runSetup();
+        }
```
Comment (avoc-ado) [inline main.ts:91]: curious, will this succeed if the internet is disconnected for a while? (like a minute)

**Review by avoc-ado — APPROVED** [review]
🔥 legend -- good stuff

**Comment by avoc-ado** [comment]
i think richard needs to merge it. no codeowners file and i don't have permission here

**Review by dbvora03 — APPROVED** [review]
noice

File: main.ts (hunk 1 above)
Comment (leantOnSol) [reply main.ts:91]: Yup, tried w/ simulating a dc by just pulling the LAN cable lol

Comment (dbvora03) [reply main.ts:91]: i laughed out loud

**Review by richardwu — APPROVED** [review]

**Pull request merged**